src/auth-service/config/firebase-admin.js

```diff
@@ -2,8 +2,50 @@ const admin = require("firebase-admin");
 const constants = require("./constants");
 const { initializeApp } = require("firebase-admin/app");
 const serviceAccount = require(`${constants.GOOGLE_APPLICATION_CREDENTIALS}`);
+const functions = require("firebase-functions");
+const joinUtil = require("../utils/join");
+const { logObject } = require("../utils/log");
 
 initializeApp({
   credential: admin.credential.cert(serviceAccount),
   databaseURL: constants.FIREBASE_DATABASE_URL,
 });
+
+exports.newUserSignUp = functions.auth.user().onCreate(async (user) => {
+  try {
+    const email = user.email;
+    const message = `Welcome to AirQo, ${user.firstName}`;
+    const subject = "Welcome to AirQo!";
+
+    admin.firestore().collection("users").doc(user.uid).set({
+      email: user.email,
+    });
+
+    return await joinUtil.sendFeedback({
+      email,
+      message,
+      subject,
+    });
+  } catch (error) {
+    logObject("error", error);
+  }
+});
+
+exports.userDeleted = functions.auth.user().onDelete(async (user) => {
+  try {
+    const email = user.email;
+    const message = `AirQo Account Deleted, ${user.firstName}`;
+    const subject = "AirQo Account Successfully Deleted!";
+
+    const doc = admin.firestore().collection("users").doc(user.uid);
+    doc.delete();
```

Comment:
This is wonderful @Baalmart . Just a few enhancements.
Below are list of documents related to the user that we shall delete.

```
admin.firestore().collection("users").doc(user.uid);
    doc.delete();
admin.firestore().collection("kya").doc(user.uid);
    doc.delete();
admin.firestore().collection("analytics").doc(user.uid);
    doc.delete();
admin.firestore().collection("notifications").doc(user.uid);
    doc.delete();
admin.firestore().collection("favourite-places").doc(user.uid);
    doc.delete();
```

**NB:** 
1. The actual collection names shall be shared with you due to security reasons @Baalmart . Consider using environment variables to store the collection names.
2. You might want to add error handling while deleting each document because some users might not have all the documents.

Reply:
Thanks @NoahNsimbe for this.
This is going to be addressed asap.

I am struggling to locate the private message you shared, I guess i would have to just reach out again.